cpu/xmc1000/include/periph/usic_defs.h

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2015 Sebastian Sontberg <sebastian@sontberg.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup         cpu_xmc1000
+ * @{
+ *
+ * @file
+ * @brief           Code shared by peripheral drivers using the USIC
+ *
+ * @author          Sebastian Sontberg <sebastian@sontberg.de>
+ *
+ * @}
+ */
+
+#ifndef USIC_DEFS_H_
+#define USIC_DEFS_H_
+
+#include "periph_cpu.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup cpu_xmc1000_usic USIC configuration
+ * @ingroup cpu_xmc1000
+ * @{
+ */
+
+#define USIC_CTQIN_PDIV  (0) /**< f<sub>CTQIN</sub> is provided by f<sub>PDIV</sub> */
+#define USIC_CTQIN_PPP   (1) /**< f<sub>CTQIN</sub> is provided by f<sub>PPP</sub>  */
+#define USIC_CTQIN_SCLK  (2) /**< f<sub>CTQIN</sub> is provided by f<sub>SCLK</sub> */
+#define USIC_CTQIN_MCLK  (3) /**< f<sub>CTQIN</sub> is provided by f<sub>MCLK</sub> */
+
+#define USIC_MODE_ASC    (0) /**< select ASC/UART mode */
+#define USIC_MODE_SSC    (1) /**< select SSC/SPI mode  */
+
+/**
+ * @brief Struct to hold the configuration values necessary to
+ *        initialize a USIC channel 
+ * @{
+ */
+typedef struct {
+    uint32_t ccr;      /**< channel control register           */
+    uint32_t sctr;     /**< shift control register             */
+    uint32_t tcsr;     /**< transmit control/status register   */
+    uint32_t pcr;      /**< protocol control register          */
+    uint32_t inpr;     /**< interrupt node pointer register    */
+    uint8_t dx2_dsel;  /**< data selection for input stage DX2 */
+} usic_controls_t;
+/** @} */
+
+/**
+ * @brief   USIC BRG (baud rate generator) register configuration
+ * @{
+ */
+typedef struct {
+    union {
```

Comment:
I would like to keep them. I watched the build sizes *very* closely when I introduced them and code size actually decreased slightly while they increase readability a lot. Maybe because they are const and never read out again (I access the uint32_t union only after setting them). If you show me that clang fucks this up, then I'll change it. But gcc is smart enough (at least 5.2).